controllers/etcdcluster_controller.go

```diff
@@ -86,51 +115,131 @@ func (r *EtcdClusterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error)
-		return ctrl.Result{}, nil
+		return ctrl.Result{}, &event.ServiceCreatedEvent{Object: cluster, ServiceName: service.Name}, nil
 	}
 	log.V(2).Info("Service exists", "service", service.Name)
 
-	peers := &etcdv1alpha1.EtcdPeerList{}
-	if err := r.List(ctx, peers, client.MatchingFields{clusterNameSpecField: cluster.Name}); err != nil {
-		log.Error(err, "unable to list peers")
-		return ctrl.Result{}, err
-	}
-
-	// Assume we can't communicate with the cluster
-	log.V(1).Info("Unable to communicate with etcd cluster")
-	if cluster.Spec.Replicas != nil && int32(len(peers.Items)) < *cluster.Spec.Replicas {
-
-		// Create more peers
-		peerName := nextAvailablePeerName(cluster, peers.Items)
-		log.V(1).Info("Insufficient peers for replicas, adding new peer",
-			"current-peers", len(peers.Items),
-			"desired-peers", cluster.Spec.Replicas,
-			"peer", peerName)
-		peer := peerForCluster(cluster, peerName)
-		if err := r.Create(ctx, peer); err != nil {
-			log.Error(err, "Failed to create peer",
+	if members == nil {
+		peers := &etcdv1alpha1.EtcdPeerList{}
+		if err := r.List(ctx, peers, client.MatchingFields{clusterNameSpecField: cluster.Name}); err != nil {
+			log.Error(err, "unable to list peers")
+			return ctrl.Result{}, nil, err
+		}
+
+		if cluster.Spec.Replicas != nil && int32(len(peers.Items)) < *cluster.Spec.Replicas {
+			// Create more peers
+			peerName := nextAvailablePeerName(cluster, peers.Items)
+			log.V(1).Info("Insufficient peers for replicas, adding new peer",
+				"current-peers", len(peers.Items),
+				"desired-peers", cluster.Spec.Replicas,
 				"peer", peerName)
-			return ctrl.Result{}, err
+			peer := peerForCluster(cluster, peerName)
+			if err := r.Create(ctx, peer); err != nil {
+				log.Error(err, "Failed to create peer",
+					"peer", peerName)
+				return ctrl.Result{}, nil, err
+			}
+			return ctrl.Result{}, &event.PeerCreatedEvent{Object: cluster, PeerName: peer.Name}, nil
 		}
-		return ctrl.Result{}, nil
+	} else {
+		log.V(2).Info("Cluster communication established")
 	}
+	return ctrl.Result{RequeueAfter: time.Second * 10}, nil, nil
+}
 
-	return ctrl.Result{}, nil
+// +kubebuilder:rbac:groups=etcd.improbable.io,resources=etcdclusters,verbs=get;list;watch
+// +kubebuilder:rbac:groups=etcd.improbable.io,resources=etcdclusters/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=core,resources=services,verbs=get;list;watch;create
+// +kubebuilder:rbac:groups=etcd.improbable.io,resources=etcdpeers,verbs=get;list;watch;create
+// +kubebuilder:rbac:groups=core,resources=endpoints,verbs=get;list
+
+func (r *EtcdClusterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	log := r.Log.WithValues("cluster", req.NamespacedName)
+
+	cluster := &etcdv1alpha1.EtcdCluster{}
+	if err := r.Get(ctx, req.NamespacedName, cluster); err != nil {
+		log.Error(err, "unable to fetch EtcdCluster resource")
```

Comment:
These errors get logged again, by the controller-runtime, so I decided not to bother logging them in the Reconcile function. 
And I set things up so that those logged errors get included in the test results.
So you could consider removing this.